supplementary_style_guide/style_guidelines/general-formatting.adoc

```diff
@@ -100,5 +100,67 @@ Text for note.
 ====
 ----
 
-// [[product-names-versions-ref]]
-// == Product names and version references
+[product-names-versions-ref]]
+== Product names and version references
+
+Whenever you refer to the name of your product in full, or in its abbreviated form, or when you refer to the major and minor version of your product, avoid using hard-coded references and use attributes instead.
+Only use hard-coded version references if the version that you are referring to in a particular case never changes.
+
+=== Attribute file
+
+Define attributes for product name and product version and store them in a dedicated attributes file for each set of product documentation.
+For examples of where you can store the shared attributes file inside your documentation repository, see the link:https://github.com/redhat-documentation/modular-docs/tree/mod-doc-repo-example[Example modular documentation repository].
+Include the attributes file at the beginning of the `master.adoc` files of all titles in your documentation set:
+
+.master.adoc
+[source,asciidoc,options="nowrap"]
+----
+\include::<path-to-directory-with-attributes-file>/attributes.adoc[]
+----
+
+=== Minimum required attributes
+
+Define attributes for the following values in each documentation set.
+Note that the attribute names used in this section are only meant as examples.
+You can use different attribute names:
+
+The name of the product:: Use the product name attribute for all instances of the product name where possible.
+Avoid using hard-coded product names.
+For example:
++
+.Example product name attribute
+----
+:name-product: Red Hat build of Quarkus
+----
+
+The abbreviated form of the product name:: If it is necessary for your product, you can use an attribute to store a shortened version of the name of your product, for example:
++
+.Example abbreviated product name attribute
+----
+:name-product-abbreviated: RHBQ
+----
+
+The major and minor version of the product:: Use an attribute for the product version in cases where the product version can change with each release and the content is still correct.
+For example:
++
+.Example long and short product version attributes
+----
+:version-product-long: 1.11.6
```

Comment:
Probably nitpicky, but I'd rather see these called minor and major, as opposed to long/short. Long/short makes sense to me in terms of the name, but not necessarily for version numbers.

Reply:
OK, but how about _minor_ and _patch_? Technically speaking, _major_ would correspond to the first digit in semantic versioning terminology, and minor would correspond to the second digit. Although it is not  a typical case, some products might need a major version attribute, too, so following the semantic versioning terminology would help avoid potential naming conflicts, WDYT?